### DISCUSSION TITLE
Bootstrap class styles not appeding while getting getCss()

### DISCUSSION THREAD

**Discussion by senthil177cse** [GrapesJS/grapesjs#4262] [Show and tell]
i have tried to generate css in JSON but i am not getiing bootstrap css while export the getCSS()..
i have attached my screenshot here you can see the "selectors":["col-12"] col-12 is bootstrap class for this bootstrap css style not getting while getCss()...
below i have lis my code....

command(editor) {
editor.Modal.setTitle('CSS JSON')
                      .setContent(`<textarea style="width: 100%; height: 250px;">
                      ${JSON.stringify(editor.getCss({json: true}))}
                      </textarea>`)
                      .open();
                  
![image (58)](https://user-images.githubusercontent.com/50009624/163564403-882cb683-f59c-44ff-b235-d1b3f9e2cbdb.jpg)
}

**Comment by senthil177cse** [comment]
@artf  any update for this please......
i am waiting for your reply i have urgent basis resolve this issue pls...

**Reply by artf** [reply]
External styles are not imported/parsed by the editor, those are only rendered in the canvas.

**Reply by senthil177cse** [reply]
already i have imported the styles check my screenshot...
![image](https://user-images.githubusercontent.com/50009624/164172012-1bde0098-d5db-4144-b778-ada81f565844.png)

**Reply by senthil177cse** [reply]
Hi @artf.... its very urgent pls respond.... 
we have to show some demo to management....

**Reply by artf** [reply]
This is what I meant for external styles...

**Reply by senthil177cse** [reply]
okay.... yaeh after imported styles are appearing in canvas but while convert editor.getCss() bootstrap css styles not fetching...

**Reply by senthil177cse** [reply]
@artf  Pls Expalin clearly because i am not getting properly...
what ever i have shared screenshot is correct one or wrong one? if wrong what i need to do...

**Reply by senthil177cse** [reply]
Hi @artf  any update on this?.. please i need to show some demo of this

**Reply by lexoyo** [reply]
Hi
Are you sure this isn't the normal behavior?
Please provide the bug report info (when you create a new issue, it has questions about the bug you are facing)

**Reply by lexoyo** [reply]
Also keep in mind we're volunteer programmers like you

Have a great weekend!

**Reply by senthil177cse** [reply]
atleast you can explain how the bootstrap styles can render in getCss()..
how to parse the external css using grapesjs?

**Reply by lexoyo** [reply]
I don't think it is possible, I don't see how

**Reply by senthil177cse** [reply]
@artf  can you give solution for this please?

**Reply by ronaldohoch** [reply]
@senthil177cse i think you're tryng to do it in wrong way. The full bootstrap file shouldn't goes in cssJson, but the import should added in the final html.

**Reply by senthil177cse** [reply]
@ronaldohoch 
i dont want to go full bootstrap css for ex: if i am adding button <button class="btn btn-success">Test</button>
here btn and btn-success bootstrap css need to add in cssJson right?

**Reply by ronaldohoch** [reply]
@senthil177cse 
Grapesjs and as far i know, no other framework does tree-shaking to css code :/

**Reply by Ju99ernaut** [reply]
You could try loading the bootstrap css as a string during init, that way the style manager has access to it. You're currently loading bootstrap css through links so the editor has no access to it, `getCss` only returns css that has been parsed by the editor.

 However this might cause performance issues, once tried it for tailwindcss but it was too large and the editor failed to parse it. Other than this you'd probably need to remove unused css manually.
